File: src/main/java/com/yui/models/Category.java
package com.yui.models;

import java.sql.Timestamp;
import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name="categories")
public class Category {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private int id;
    private String display;
    private Timestamp createdAt;
    
    @OneToMany(mappedBy="category", fetch=FetchType.EAGER)
    private List<Artwork> artworks;
}

File: src/entity/Category.ts
import {
    Column,
    Entity,
    ManyToOne,
    PrimaryGeneratedColumn,
    Timestamp,
} from 'typeorm';
import { Artwork } from '.';

@Entity()
export class Category {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    display: string;

    @Column()
    createdAt: Timestamp;

    @ManyToOne(() => Artwork, (artwork) => artwork.category)
    artworks: Artwork[];
}
